src/main/java/graphql/nadel/engine/NadelExecutionStrategy.java

```diff
@@ -153,28 +149,24 @@ public NadelExecutionStrategy(List<Service> services,
             // take the original query and transform it into the underlying query needed for that top level field
             //
             GraphQLSchema underlyingSchema = service.getUnderlyingSchema();
-            CompletableFuture<QueryTransformationResult> queryTransformCF = queryTransformer
+            CompletableFuture<QueryTransformationResult> transformedQueryCF = queryTransformer
                     .transformMergedFields(executionContext, underlyingSchema, operationName, operation, singletonList(mergedField), serviceExecutionHooks, service, serviceContext);
 
-            resultNodes.add(queryTransformCF.thenCompose(queryTransform -> {
+            resultNodes.add(transformedQueryCF.thenCompose(transformedQuery -> {
+                Map<String, FieldTransformation> fieldIdToTransformation = transformedQuery.getFieldIdToTransformation();
+                Map<String, String> typeRenameMappings = transformedQuery.getTypeRenameMappings();
 
-                Map<String, FieldTransformation> fieldIdToTransformation = queryTransform.getFieldIdToTransformation();
-                Map<String, String> typeRenameMappings = queryTransform.getTypeRenameMappings();
+                ExecutionContext newExecutionContext = buildServiceVariableOverrides(executionContext, transformedQuery.getVariableValues());
 
-                ExecutionContext newExecutionContext = buildServiceVariableOverrides(executionContext, queryTransform.getVariableValues());
-
-                String topLevelFieldId = NodeId.getId(esi.getFieldDefinition());
-                Optional<GraphQLError> maybeTopLevelFieldError = queryTransform.getRemovedFieldMap()
-                        .getRemovedFieldById(topLevelFieldId)
-                        .map(NormalizedFieldAndError::getError);
-                boolean topLevelFieldExecutionShouldBeSkipped = maybeTopLevelFieldError.isPresent();
-                if (topLevelFieldExecutionShouldBeSkipped) {
+                Optional<GraphQLError> maybeTopLevelFieldError = getTopLevelFieldError(esi, transformedQuery);
+                boolean skipTopLevelFieldExecution = maybeTopLevelFieldError.isPresent();
+                if (skipTopLevelFieldExecution) {
                     GraphQLError topLevelFieldError = maybeTopLevelFieldError.get();
-                    return CompletableFuture.completedFuture(getSkippedServiceCallResult(nadelContext, esi, executionContext, topLevelFieldError));
+                    return CompletableFuture.completedFuture(getRejectedTopLevelFieldResult(nadelContext, esi, topLevelFieldError));
```

Comment:
the naming is off here:
They are either forbidden or rejected. Choose one and make it consistent :-)